src/main/c/waf_jni.c

```diff
@@ -1140,6 +888,182 @@ static bool _cache_single_class_weak(JNIEnv *env, const char *class_name,
     return true;
 }
 
+static ddwaf_builder _get_builder_checked(JNIEnv *env, jclass clazz,
+                                          jobject builder_obj)
+{
+    UNUSED(clazz);
+
+    ddwaf_builder builder = (ddwaf_builder) (intptr_t) JNI(
+            GetLongField, builder_obj, _builder_ptr);
+    if (JNI(ExceptionCheck)) {
+        return NULL;
+    }
+
+    if (!builder) {
+        JNI(ThrowNew, jcls_rte, "The Builder has already been cleared");
+    }
+
+    return builder;
+}
+
+JNIEXPORT jboolean JNICALL Java_com_datadog_ddwaf_WafBuilder_removeConfigNative(
+        JNIEnv *env, jclass clazz, jobject builder, jstring path)
+{
+    UNUSED(clazz);
+    UNUSED(env);
+    if (builder && path) {
+        ddwaf_builder ddwaf_builder = _get_builder_checked(env, clazz, builder);
+        if (JNI(ExceptionCheck)) {
+            JAVA_LOG(DDWAF_LOG_DEBUG, "build not found");
+            return JNI_FALSE;
+        }
+        const char *path_string = JNI(GetStringUTFChars, path, NULL);
+        if (JNI(ExceptionCheck)) {
+            JAVA_LOG(DDWAF_LOG_DEBUG, "path could not be converted for ddwaf");
+            return JNI_FALSE;
+        }
+        int path_length = JNI(GetStringLength, path);
+        if (JNI(ExceptionCheck)) {
+            JAVA_LOG(DDWAF_LOG_DEBUG,
+                     "path length could not be found for ddwaf");
+            return JNI_FALSE;
```

Comment:
This leaks, release string is not called, you can use `goto` for error handling.